src/main/TreeQuery.cs

```diff
@@ -0,0 +1,200 @@
+                        if (!string.IsNullOrEmpty(regionIdStr) && Guid.TryParse(regionIdStr, out Guid parsedGuid))
+                        {
+                            regionId = parsedGuid;
+                        }
+
+                        result = new TreeQuery
+                        {
+                            Postsynaptics = TreeQueryParseGuidArray(array.GetQueryArrayOrDefault(typeFromHandle.GetQueryKey(nameof(Postsynaptics)))),
+                            RegionId = regionId,
+                            DirectionValues = array.GetNullableEnumValue<DirectionValues>(typeFromHandle.GetQueryKey(nameof(DirectionValues))),
+                            AvatarUrl = HttpUtility.UrlDecode(TreeQueryGetQuerySingleValue(array, typeFromHandle.GetQueryKey(nameof(AvatarUrl)))),
+                            ExpandUntilPostsynapticMirrors = TreeQueryParseGuidArray(array.GetQueryArrayOrDefault(typeFromHandle.GetQueryKey(nameof(ExpandUntilPostsynapticMirrors))))
+                        };
+
+                        result2 = true;
+                    }
+                    catch (ArgumentException)
+                    {
+                        // Handle invalid enum values gracefully
+                        result = null;
+                        result2 = false;
+                    }
+                }
+            }
+
+            return result2;
+        }
+
+        private static Dictionary<string, List<string>> TreeQueryParseQueryString(string queryString)
+        {
+            var parameters = new Dictionary<string, List<string>>(StringComparer.OrdinalIgnoreCase);
+            string[] pairs = queryString.Split('&');
+
+            foreach (string pair in pairs)
+            {
+                if (string.IsNullOrEmpty(pair)) continue;
+
+                int equalIndex = pair.IndexOf('=');
+                if (equalIndex > 0)
+                {
+                    string key = HttpUtility.UrlDecode(pair.Substring(0, equalIndex));
+                    string val = HttpUtility.UrlDecode(pair.Substring(equalIndex + 1));
+
+                    if (!parameters.ContainsKey(key))
+                        parameters[key] = new List<string>();
+                    parameters[key].Add(val);
+                }
+                else if (equalIndex == 0)
+                {
+                    // Handle case where key is empty but there's a value
+                    continue;
+                }
+                else
+                {
+                    // Handle case where there's no equals sign (key without value)
+                    string key = HttpUtility.UrlDecode(pair);
+                    if (!parameters.ContainsKey(key))
+                        parameters[key] = new List<string>();
+                    parameters[key].Add("");
+                }
+            }
+
+            return parameters;
+        }
+
+        private static List<string> TreeQueryConvertParametersToArray(Dictionary<string, List<string>> parameters)
```

Comment:
Is this method necessary? Can we use SelectMany to do this?

https://asusualcoding.wordpress.com/2024/06/03/how-to-flatten-the-nested-lists-using-linq-in-c/#:~:text=If%20we%20want%20to%20merge,resulting%20sequences%20into%20a%20list.&text=The%20above%20code%20snippet%20will,to%20a%20single%20numeric%20list.

Reply:
Done with another approach